behaviors/TranslatablePageUrl.php

```diff
@@ -172,4 +173,34 @@ public function setViewBagUrlAttributeTranslated($value, $locale)
     {
         $this->setSettingsUrlAttributeTranslated($value, $locale);
     }
+
+    /**
+     * Handler for the pages.menuitem.resolveItem event.
+     */
+    public static function resolveMenuItem($item, $url, $theme)
+    {
+        $result = null;
+
+        if ($item->type == 'cms-page') {
+            if (!$item->reference) {
+                return;
+            }
+
+            $page = \Cms\Classes\Page::loadCached($theme, $item->reference);
+            $translate = Translator::instance();
+            if ($page->hasTranslatablePageUrl($translate->getLocale())) {
+                $page->rewriteTranslatablePageUrl($translate->getLocale());
+            }
+            $pageUrl = $translate->getPathInLocale($page->url, $translate->getLocale());
```

Comment:
You're using "translate->getLocale()" 3 times in a row... please, assign the result to a variable and reuse variable.